semgrep/semgrep/evaluation.py

```diff
@@ -99,7 +126,8 @@ def _evaluate_single_expression(
                 f"at least one rule needs to execute arbitrary code; this is dangerous! if you want to continue, enable the flag: {RCE_RULE_FLAG}",
                 code=NEED_ARBITRARY_CODE_EXEC_EXIT_CODE,
             )
-        assert expression.operand, "must have operand for this operator type"
+        if not isinstance(expression.operand, str):
+            raise InvalidRuleSchemaError("pattern-where-python must have a string value")
```

Comment:
Is this the correct error message? What does this case have to do with `pattern-where-python`?

Reply:
This was more of a code hygiene fix since `operand` is no longer guaranteed to be a string, so this appeases mypy. Also, when Python is run in optimize mode `assert` statements are removed, so we shouldn't rely on them:

```
$ python --help
...
-O     : remove assert and __debug__-dependent statements; add .opt-1 before
         .pyc extension; also PYTHONOPTIMIZE=x
-OO    : do -O changes and also discard docstrings; add .opt-2 before
         .pyc extension
...
```